internal/host/transition.go

```diff
@@ -665,6 +667,38 @@ func (th *transitionHandler) HasInstallationInProgressTimedOut(sw stateswitch.St
 	return time.Since(time.Time(sHost.host.Progress.StageUpdatedAt)) > maxDuration, nil
 }
 
+func (th *transitionHandler) PostHostPreparationTimeout() stateswitch.PostTransition {
+	ret := func(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
+		sHost, ok := sw.(*stateHost)
+		if !ok {
+			return errors.New("PostHostPreparationTimeout incompatible type of StateSwitch")
+		}
+		params, ok := args.(*TransitionArgsRefreshHost)
+		if !ok {
+			return errors.New("PostRefreshHost invalid argument")
+		}
+		var (
+			err error
+		)
+		template := statusInfoPreparationTimeout
+		failingConditons := []string{}
+		if !params.conditions["installation-disk-speed-check-successful"] {
+			failingConditons = append(failingConditons, statusInfoPreparationTimeoutDiskSpeed)
+		}
+		if !params.conditions["successful-container-image-availability"] {
+			failingConditons = append(failingConditons, statusInfoPreparationTimeoutImageAvailability)
+		}
+		template = strings.Replace(template, "$FAILING_CONDITIONS", strings.Join(failingConditons, "\n"), 1)
```

Comment:
```suggestion
		statusInfo := strings.Replace(statusInfoPreparationTimeout, "$FAILING_CONDITIONS", strings.Join(failingConditons, "\n"), 1)
```

Reply:
Here for example:

https://github.com/openshift/assisted-service/blob/master/internal/host/statemachine.go#L536-L550

This is a transition on error to "insufficient" which works in the same way as far as I can tell?

Reply:
I believe that this is the transition that then moves to known

https://github.com/openshift/assisted-service/blob/master/internal/host/statemachine.go#L918-L938

I was under the impression that this was already well understood?

Reply:
They are not the same:
The first has condition `stateswitch.Not(If(SufficientOrUnknownInstallationDiskSpeed)))`
And the second has the opposite condition:
`If(SufficientOrUnknownInstallationDiskSpeed)`  (in `isSufficientForInstall`)